### ISSUE TITLE
feat: Copy Entire Thread Button

### ISSUE THREAD

**Issue by pbeens** [janhq/jan#1459]
**Problem**
Currently you can only individually copy (with the copy button) the User fields or the Assistant fields, but not the entire thread.

**Success Criteria**
I would love to see a button that lets you copy the entire thread for pasting into another program.

**Additional context**
This could be done in the Threads Setting menu or via a button at the top and/or the bottom of the thread.

**Comment by 0xSage** [comment]
I know this isn't the perfect immediate solution, but you already have all the thread history in your own files: 

![Export-1704965101591](https://github.com/janhq/jan/assets/69952136/fdd639e4-75dc-44b9-ac9c-0cbf9d25ade4)

Adding this a future feature enhancemnt request :)

**Comment by pbeens** [comment]
Thanks. This Python script will extract what I need. Hopefully others will find this useful until Jan is updated its own solution. 

```
import json

# Replace filename with the path to your JSONL file
file = r'C:\Users\Peter\jan\threads\jan_1704487430\messages.jsonl'

# Open the file and read it line by line
with open(file) as f:
    for line in f:
        # Load each line as a JSON object
        data = json.loads(line)

        # Extract the role and content fields from the JSON object
        role = data.get('role')
        content = data.get('content')[0].get('text').get('value')

        # Print the role and content fields
        print(f"Role: {role}\nContent: {content}\n")
```

**Comment by imtuyethan** [comment]
Thanks for your suggestion. We will continue to improve Jan threads experience. This request will be included in #1940 which belongs to epic: #1507
